azurerm/internal/services/loganalytics/validate/log_analytics_name.go

```diff
@@ -0,0 +1,53 @@
+package validate
+
+import (
+	"encoding/base64"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+func LogAnalyticsGenericName(i interface{}, k string) (warnings []string, errors []error) {
```

Comment:
since this is an helper function can we make this internal:

```suggestion
func logAnalyticsGenericName(i interface{}, k string) (warnings []string, errors []error) {
```

Reply:
This is actually doubled up, both exist, I'll remove the exported func.